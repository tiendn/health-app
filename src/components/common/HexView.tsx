import Image from "next/image";
import React from "react";

import styles from "./style.module.scss";
import { inter } from "@/theme/fonts";
import clsx from "clsx";
import { Icons } from "@/theme/image";
import { capitalizeFirstLetter } from "@/utils/helper";

export const HexViewType = {
	morning: Icons.knife,
	lunch: Icons.knife,
	dinner: Icons.knife,
	snack: Icons.cup,
};

interface Props {
	type: keyof typeof HexViewType;
	onClick: () => void;
}

const HexView = ({ type = "morning", onClick }: Props) => {
	return (
		<div onClick={onClick} className={styles.hexViewContainer}>
			<Image width={56} height={56} src={HexViewType[type]} alt="icon" />
			<span className={clsx(styles.hexViewText, inter.className)}>
				{capitalizeFirstLetter(type)}
			</span>
		</div>
	);
};

export default HexView;
