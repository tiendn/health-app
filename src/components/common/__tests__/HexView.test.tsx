import React from "react";
import renderer from "react-test-renderer";
import HexView from "../HexView";

test("Badge renders correctly", () => {
	const tree = renderer.create(<HexView type="morning" onClick={() => {}} />).toJSON();
	expect(tree).toMatchSnapshot();
});
