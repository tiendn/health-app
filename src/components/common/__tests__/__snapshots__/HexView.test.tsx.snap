// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders correctly 1`] = `
<div
  className="hexViewContainer"
  onClick={[Function]}
>
  <img
    alt="icon"
    data-nimg="1"
    decoding="async"
    height={56}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=%2Fimages%2Ficons%2Fknife.png&w=128&q=75"
    srcSet="/_next/image?url=%2Fimages%2Ficons%2Fknife.png&w=64&q=75 1x, /_next/image?url=%2Fimages%2Ficons%2Fknife.png&w=128&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    width={56}
  />
  <span
    className="hexViewText className"
  >
    Morning
  </span>
</div>
`;
